packages/vscode-extension/src/project/metro.ts

```diff
@@ -211,18 +211,8 @@ export class Metro implements Disposable {
   }
 
   public async reload() {
-    const appReady = new Promise<void>((resolve) => {
-      const waitForAppReady = (event: string) => {
-        if (event === "RNIDE_appReady") {
-          this.devtools.removeListener(waitForAppReady);
-          resolve();
-        }
-      };
-      this.devtools.addListener(waitForAppReady);
-    });
-
     await fetch(`http://localhost:${this._port}/reload`);
-    await appReady;
+    await this.devtools.appReady();
```

Comment:
this changes the behavior. Before we'd start listening before the fetch and now this is reversed. The difference is that fetch is also async, but because of that it may already be too late to start listening for event after fetch is complete, becuase the event might've already be sent and we may miss it